languages/ruby/exercises/concept/booleans/.docs/instructions.md

```diff
@@ -0,0 +1,30 @@
+Continuing your work with the amusement park, you are tasked with writing some utility methods to facilitate checking an attendee can use a ride.
+
+## 1. Check if an attendee has a ride pass
+
+Implement the `has_pass?` method to return a boolean (`true`/`false`) value based on the presence of a ride pass.
```

Comment:
The `has_pass?` name is very close to `is_pass?` and query methods have the question mark in order to ask a question.  `pass?` may be clear enough, if it is `nil` there is none`, if it is `false` they have one that is invalid, and if it is something else, it is presumed to be valid.

Reply:
```suggestion
Implement the `pass?` method to return a boolean (`true`/`false`) value based on the presence of a ride pass.
```

Reply:
So, (sorry to disagree - I feel like I'm doing that lots!) I think this should be `has_pass?`. The reason is because this isn't asking about the state of the object itself (like `happy?` rather than `is_happy?`) it's asking about its possession of something else, and because pass can be a verb or a noun, and so I feel that the `has_` has value. (e.g. the semantic meanings of "Does this attendee have a pass?" is clearer to me than "Does this attendee pass?" which is how I would interpret `pass?`).

To be clear, I don't mind either way, but if I was writing the code I would write `has_pass?` :)

Reply:
I get what you are saying, it is the meaning in both ways that it does match, you can use it for both.  That is the beauty of it, really.  In either context it would work while reading the code.

I do see the `has_` along with the question mark used often in Rails.  Of course, I sometimes agree when it has a very specific use, but don't agree if it can be used in the more general sense.  In this case, it answers, both questions, yes they _have_ a pass, and they _may_ pass.  And the getter `pass` can be used to provide the data of the pass value itself as well as the boolean context to be used for the query method, provided that the double negation doesn't happen in the implementation of `pass?` or `is_pass?` or `has_pass?`.  If we do name `has_pass?` then it may follow that we should name `is_pass?` to validate their pass.  That is does not need to be the case.

In the end, of course, it will work, the reader will be biased in one way or another based on what it communicates and what their experience is.  Will they know without the `is_` and `has_` names that they can apply it in all three ways from the name?  Or will the student realize that they can do so?  A lot of moving parts when it comes to code, even more when it comes to interpretation.

Reply:
Yeah, all the complexity is why I'd just be explicit if I was naming this in the real world. effectively the three possibilities are: `has_pass?`, `can_pass?` or `is_a_pass?` (maybe `is_pass?`). With words that can be nouns, adjectives and verbs, I always prefer to be explicit in my naming. For works that can only be one I'd be more inclined to drop the prefix (e.g. if the object was `lanyard`) (although in the case of the object "having" something that wasn't directly a property of itself, I'd probably still prefer `has_...?` - e.g. I'd consider model an attendee with `attendance_date?` but `has_lanyard?`)